src/librustc_traits/dropck_outlives.rs

```diff
@@ -80,22 +80,26 @@ fn dropck_outlives<'tcx>(
             let mut fulfill_cx = TraitEngine::new(infcx.tcx);
 
             let cause = ObligationCause::dummy();
+            let mut constraints = DtorckConstraint::empty();
             while let Some((ty, depth)) = ty_stack.pop() {
-                let DtorckConstraint {
-                    dtorck_types,
-                    outlives,
-                    overflows,
-                } = dtorck_constraint_for_ty(tcx, DUMMY_SP, for_ty, depth, ty)?;
+                info!("{} kinds, {} overflows, {} ty_stack",
+                    result.kinds.len(), result.overflows.len(), ty_stack.len());
+                dtorck_constraint_for_ty(tcx, DUMMY_SP, for_ty, depth, ty, &mut constraints)?;
 
                 // "outlives" represent types/regions that may be touched
                 // by a destructor.
-                result.kinds.extend(outlives);
-                result.overflows.extend(overflows);
+                result.kinds.extend(constraints.outlives.drain(..));
+                result.overflows.extend(constraints.overflows.drain(..));
+
+                // At some point we just need to stop
```

Comment:
I haven't fully brought all this back into my mental cache; can you say more here on why `>= 1` is the right threshold?

If all you want to do is stop as soon as there is *any* overflow (which is how I interpret this), then fine. But the comment as written makes it sound like the threshold is a more interesting value, like 100 or something.

Reply:
(Anyway apart from that, I think these changes look fine.)

Reply:
Ah, yeah, I can probably update the comment -- I originally had this at 4, but realized that for many cases (if not all) where we overflow we're going to end up with such large types that even printing one might not be successful (since printing is so slow).

Reply:
This conditional is basically because with the new code we detect thousands (and then millions.. and so on) of overflows, eventually leading to an OOM on the `Perfect` enum; I initially thought maybe we should have some small number, like 10, here -- but then in thinking more, I had decided that chances are, the first overflow contains all the information you need, and there's no reason for us to spin further.

I can update this comment with that summary?